include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,533 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+#include <vk_mem_alloc.h>
+
+#include "utils/misc.h"
+#include "vkwrap/core.h"
+#include "vkwrap/error.h"
+
+#include <range/v3/algorithm/any_of.hpp>
+
+#include <range/v3/range/conversion.hpp>
+
+#include <array>
+#include <bit>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+namespace std
+{
+
+template <> struct hash<vk::Buffer>
+{
+    size_t operator()( const vk::Buffer& buffer ) const
+    {
+        return std::bit_cast<size_t>( buffer.operator VkBuffer() );
```

Comment:
This is equivalent still to:
```cpp
return std::bit_cast<size_t>(*buffer);
```

Reply:
There is no `operator *` in `vk::Buffer`.